src/pages/Intents/CustomIntents.js

```diff
@@ -0,0 +1,80 @@
+import { useDataMutation } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import isEmpty from 'lodash/isEmpty'
+import isEqual from 'lodash/isEqual'
+import React, { useEffect, useMemo, useState } from 'react'
+import { useIsAuthorized } from '../../auth'
+import FooterStripButtons from '../../components/footerStripButton/FooterStripButtons'
+import { CustomIntentInfo } from '../../components/noticeAlert'
+import Page from '../../components/page/Page'
+import CustomIntentsList from './CustomIntentsList'
+import {
+    saveCustomIntentsMutation,
+    useReadCustomIntentsDataStore,
+} from './intentsDatastoreQuery'
+import { IntentsList } from './mockList'
+
+//TODO: change IntentsList mock data to real data
+const CustomIntents = () => {
+    const { hasAuthority } = useIsAuthorized()
+    const { customIntents } = useReadCustomIntentsDataStore()
+    const [disableSave, setDisableSave] = useState(true)
+    const [customIntentSettings, setCustomIntentSettings] =
+        useState(IntentsList)
+    const initialValues = useMemo(() => {
+        return customIntents
+    }, [customIntentSettings])
+    const [mutate, { error, data }] = useDataMutation(saveCustomIntentsMutation)
+
+    useEffect(() => {
+        initialValues &&
+        customIntentSettings &&
+        !isEqual(initialValues, customIntentSettings)
+            ? setDisableSave(false)
```

Comment:
I feel like this might just be possible as `const disabledSave = !isEqual(initialValues, customIntentSettings)` but that might depend on how much the function rerenders in the final implementation. 

Or, maybe even easier, you could keep track of initial values in a more-expanded-purpose `useCustomIntentSetttings` hook and then return a value that indicates whenever the "form state" is dirty/differs from the initial values (which would also help separate some of the UI/api logic more)